module/applications/components/inventory.mjs

```diff
@@ -365,155 +546,313 @@ export default class InventoryElement extends HTMLElement {
+   * @returns {Promise<Roll|void>}
    * @protected
    */
-  async _onAction(target, action, { event }={}) {
-    const inventoryEvent = new CustomEvent("inventory", {
-      bubbles: true,
-      cancelable: true,
-      detail: action
-    });
-    if ( target.dispatchEvent(inventoryEvent) === false ) return;
+  _onRollRecharge(entry, { event }={}) {
+    if ( entry instanceof Item5e ) return item.system.uses?.rollRecharge({ apply: true, event });
+    return entry.uses?.rollRecharge({ apply: true, event });
+  }
 
-    const { itemId } = target.closest("[data-item-id]")?.dataset ?? {};
-    const { activityId } = target.closest("[data-activity-id]")?.dataset ?? {};
-    const item = await this.getItem(itemId);
-    if ( !["create", "currency"].includes(action) && !item ) return;
-    const activity = item?.system.activities?.get(activityId);
+  /* -------------------------------------------- */
 
-    switch ( action ) {
-      case "activity-recharge":
-        return activity?.uses?.rollRecharge({ apply: true, event });
-      case "activity-use":
-        return activity?.use({ event });
-      case "attune":
-        return item.update({"system.attuned": !item.system.attuned});
-      case "create":
-        if ( this.document.type === "container" ) return;
-        return this._onCreate(target);
-      case "crew":
-        return item.update({"system.crewed": !item.system.crewed});
-      case "currency":
-        return new CurrencyManager({ document: this.document }).render({ force: true });
-      case "delete":
-        return item.deleteDialog();
-      case "duplicate":
-        return item.clone({name: game.i18n.format("DOCUMENT.CopyOf", {name: item.name})}, {save: true});
-      case "edit":
-        return item.sheet.render(true, { mode: ItemSheet5e.MODES.EDIT });
-      case "view":
-        return item.sheet.render(true, { mode: ItemSheet5e.MODES.PLAY });
-      case "equip":
-        return item.update({"system.equipped": !item.system.equipped});
-      case "expand":
-        return this._onExpand(target, item);
-      case "favorite":
-        return this.actor.system.addFavorite({type: "item", id: item.getRelativeUUID(this.actor)});
-      case "prepare":
-        return item.update({"system.preparation.prepared": !item.system.preparation?.prepared});
-      case "recharge":
-        return item.system.uses?.rollRecharge({ apply: true, event });
-      case "toggleCharge":
-        return item.update({ "system.uses.spent": 1 - item.system.uses.spent });
-      case "unfavorite":
-        return this.actor.system.removeFavorite(item.getRelativeUUID(this.actor));
-      case "use":
-        return item.use({ legacy: false, event });
+  /**
+   * Manage columns when the inventory element's inline size changes.
+   * @param {ResizeObserverEntry[]} entries
+   * @protected
+   */
+  _onResize([entry]) {
+    // TODO: Should accommodate uiScale here, but probably don't want to call game.settings.get every frame.
```

Comment:
We could grab it from the styles I suppose: `document.defaultView.getComputedStyle(this).getPropertyValue("--ui-scale")`.